st2common/st2common/constants/triggers.py

```diff
@@ -252,40 +252,64 @@
             "type": "string"
         },
         "year": {
-            "type": "integer"
+            "anyOf": [
+                {"type": "string"},
+                {"type": "integer"}
+            ],
         },
         "month": {
-            "type": "integer",
+            "anyOf": [
+                {"type": "string"},
+                {"type": "integer"}
+            ],
             "minimum": 1,
```

Comment:
```MESSAGE: 1000 is greater than the maximum of 6

Failed validating 'maximum' in schema['properties']['day_of_week']:
    {'anyOf': [{'type': 'string'}, {'type': 'integer'}, {'type': 'null'}],
     'maximum': 6,
     'minimum': 0}
```

Looks like it works as expected - minmax validation is only performed if parameter type is `integer`.

So this works as expected :)